### ISSUE TITLE
2.0.6, Segfault after short uptime

### ISSUE THREAD

**Issue by olafnorge** [fluent/fluent-bit#6492]
## Bug Report

**Describe the bug**
I am running fluent-bit as a `Deployment` in AWS EKS. After a short uptime, 10 minutes or even less, the POD gets restarted because of a segmentation fault.

**To Reproduce**
- Configuration:
```
[SERVICE]
    Daemon Off
    Flush 1
    Log_Level error
    Parsers_File parsers.conf
    Parsers_File custom_parsers.conf
    HTTP_Server On
    HTTP_Listen 0.0.0.0
    HTTP_Port 2020
    Health_Check On

[INPUT]
    Name     forward
    Listen   0.0.0.0
    Port     24224
    threaded on

[OUTPUT]
    # set the common bits for the output
    Name            s3
    Match           kube.*

    # Use single main thread to handle all the load and to prevent segmentation faults
    workers         0

    region          us-east-1
    bucket          fluent-bucket
    s3_key_format   /123456789/us-east-1/cluster-001/%Y/%m/%d/%H/$UUID.json

    # Added canned_acl for cross account ownership
    canned_acl      bucket-owner-full-control

    # setting the following values because we dont have a persistence layer (HDD)
    # https://docs.fluentbit.io/manual/pipeline/outputs/s3#using-s3-without-persisted-disk
    use_put_object  On
    upload_timeout  1m
    total_file_size 1M

```
- Log Message:
```
[2022/12/02 10:37:20] [engine] caught signal (SIGSEGV)
#0  0x562dbd9a566e      in  __mk_list_del() at lib/monkey/include/monkey/mk_core/mk_list.h:141
#1  0x562dbd9a56a5      in  mk_list_del() at lib/monkey/include/monkey/mk_core/mk_list.h:147
#2  0x562dbd9a59ab      in  flb_bucket_queue_delete_min() at include/fluent-bit/flb_bucket_queue.h:113
#3  0x562dbd9a59f8      in  flb_bucket_queue_pop_min() at include/fluent-bit/flb_bucket_queue.h:122
#4  0x562dbd9ab267      in  flb_engine_start() at src/flb_engine.c:851
#5  0x562dbd952bd9      in  flb_lib_worker() at src/flb_lib.c:629
#6  0x7ff746d99ea6      in  ???() at ???:0
#7  0x7ff746669a2e      in  ???() at ???:0
#8  0xffffffffffffffff  in  ???() at ???:0
```

**Your Environment**

* Version used: `2.0.6`
* Configuration: Official Helm Chart `fluent-bit-0.21.5`
* Environment name and version (e.g. Kubernetes? What version?): AWS EKS `v1.21.14-eks-fb459a0`
* Server type and version: Amazon Linux 2 `5.4.219-126.411.amzn2.x86_64`
* Operating System and version:
* Filters and plugins: 

**Additional context**

If I run fluent-bit with `threaded off` the segmentation faults don't seem to happen.

**Comment by olafnorge** [comment]
Another "more complex" configuartion also fails with segmentation faults:
```
[SERVICE]
    Daemon                    Off
    Flush                     1
    Log_Level                 error
    Parsers_File              parsers.conf
    Parsers_File              custom_parsers.conf
    HTTP_Server               On
    HTTP_Listen               0.0.0.0
    HTTP_Port                 2020
    Health_Check              On

    # Buffering & Storage (https://docs.fluentbit.io/manual/administration/buffering-and-storage#service-section-configuration)
    storage.path              /var/fluent-bit/state/flb-storage/
    storage.sync              normal
    storage.checksum          off
    storage.backlog.mem_limit 5M
    storage.metrics           on

[INPUT]
    Name             tail
    Path             /var/log/containers/abc-*.log, /var/log/containers/def-*.log, /var/log/containers/ghi-*.log, /var/log/containers/jkl-*.log, /var/log/containers/mno-*.log
    Exclude_Path     /var/log/containers/aggregator-fluent-bit-*_logging_fluent-bit-*.log, /var/log/containers/aws-node*.log, /var/log/containers/collector-fluent-bit-*_logging_fluent-bit-*.log, /var/log/containers/kube-proxy*.log
    DB               /var/fluent-bit/state/flb_container.db
    multiline.parser docker, cri
    Tag              kube.*
    threaded         on
    Mem_Buf_Limit    32MB
    Skip_Long_Lines  On
    Skip_Empty_Lines On
    Refresh_Interval 10
    Rotate_Wait      30

    # Buffering & Storage (https://docs.fluentbit.io/manual/administration/buffering-and-storage#input-section-configuration)
    storage.type     filesystem

[INPUT]
    name            fluentbit_metrics
    tag             internal_metrics
    scrape_interval 2
    scrape_on_start true
    threaded        on

[FILTER]
    Name                kubernetes
    Match               kube.*
    Kube_URL            https://kubernetes.default.svc.cluster.local:443
    Kube_CA_File        /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    Kube_Token_File     /var/run/secrets/kubernetes.io/serviceaccount/token
    Kube_Tag_Prefix     kube.var.log.containers.
    Merge_Log           On
    # a valid json log will be available under this key
    Merge_Log_Key       log_processed
    # don't limit the API response buffer
    Buffer_Size         0
    Use_Kubelet         true

[FILTER]
    Name   record_modifier
    Match  kube.*
    Record cluster_name cluster-001


[FILTER]
    Name           lua
    Match          kube.*
    script         /fluent-bit/scripts/filter_record.lua
    time_as_table  true
    protected_mode true
    call           add_timestamp

[OUTPUT]
    Name                     forward
    Match                    kube.*
    Host                     aggregator-fluent-bit
    Port                     24224

    # Buffering & Storage (https://docs.fluentbit.io/manual/administration/buffering-and-storage#output-section-configuration)
    storage.total_limit_size 5M

[OUTPUT]
    name            prometheus_exporter
    match           internal_metrics
    host            0.0.0.0
    port            2021
```


Logs:
```
[2022/12/02 13:53:04] [engine] caught signal (SIGSEGV)
#0  0x557c8a889dcd      in  cio_file_is_up() at lib/chunkio/src/cio_file.c:1223
#1  0x557c8a886019      in  cio_chunk_is_up() at lib/chunkio/src/cio_chunk.c:480
#2  0x557c8a887083      in  cio_stats_get() at lib/chunkio/src/cio_stats.c:55
#3  0x557c8a0dcdb3      in  flb_storage_metrics_update() at src/flb_storage.c:73
#4  0x557c8a14425d      in  flb_me_get_cmetrics() at src/flb_metrics_exporter.c:263
#5  0x557c8a21bf0c      in  scrape_metrics() at plugins/in_fluentbit_metrics/metrics.c:50
#6  0x557c8a21c05a      in  cb_metrics_collect_runtime() at plugins/in_fluentbit_metrics/metrics.c:73
#7  0x557c8a09cca1      in  input_collector_fd() at src/flb_input_thread.c:168
#8  0x557c8a09d4c2      in  engine_handle_event() at src/flb_input_thread.c:183
#9  0x557c8a09d4c2      in  input_thread() at src/flb_input_thread.c:384
#10 0x557c8a0ec6ba      in  step_callback() at src/flb_worker.c:43
#11 0x7f386e931ea6      in  ???() at ???:0
#12 0x7f386e201a2e      in  ???() at ???:0
#13 0xffffffffffffffff  in  ???() at ???:0
```

Btw, these are the only log lines I get also for the other config.

**Comment by yackushevas** [comment]
The same problem with **threaded on**.

**Comment by leonardo-albertovich** [comment]
Hi @olafnorge, I am in the process of testing the patch that should correct this issue and would really appreciate it if you could give it a try, the branch URL is https://github.com/fluent/fluent-bit/tree/leonardo-master-downstream-pr-6882-rework 

Thank you very much.

**Comment by cliles** [comment]
@leonardo-albertovich  - This branch no longer exists, unable to test.

**Comment by leonardo-albertovich** [comment]
That's because the pull request it was bound to has already been merged which means you should be able to test it by upgrading to the latest version (released yesterday).

**Comment by cliles** [comment]
Well then it doesn't look like this fixes #6901. ☹️

**Comment by leonardo-albertovich** [comment]
Uhm not really, I wasn't expecting it though since I haven't touched anything stackdriver related in a very long time.
